README.adoc

```diff
@@ -22,87 +22,69 @@ This section describes some important information about how this repository is s
+. If you are releasing a new `latest` version, submit a pull request to the `develop` branch of the link:{url-playbook}[`hazelcast/hazelcast-docs`] repository to do the following:
 +
-TIP: For guidance on writing content, see the {url-contribute}[contribution guidelines].
+- Update the `_redirects` file with your release version of IMDG.
 +
-NOTE: If a branch for your major or minor version does not exist, see <<major-and-minor-releases, Major and Minor Releases>>.
-
-. When you are ready to release, create a new release branch from the `.z` branch and name it with the full version number that you are releasing. For example, if you are releasing version 4.1.2, name the branch 4.1.2.
-
-. In the `antora.yml` file of your release branch, update the `version` field with the version of IMDG that you are working on.  For example from `4.1.2-SNAPSHOT` to `4.1.2`.
+NOTE: This file is where we alias the `latest` path in URLs.
 +
-[source,yaml]
+[source,bash]
 ----
-version: 4.1.2
+# Redirect latest imdg alias to the latest version
+/imdg/latest/*  /imdg/4.1.2/:splat 200!
 ----
-
-. Tag the latest commit in your release branch with a `v` followed by the version number.
-
-. If you have made any changes since creating your release branch, merge those changes back into the `.z` branch.
-
-. Delete your release branch.
 +
-WARNING: Do not delete the `.z` branch. This branch is for writing content for future patch releases.
-
-. In the `antora.yml` file of the `.z` branch, increment the `version` field. For example, `4.1.2-SNAPSHOT` becomes `4.1.3-SNAPSHOT`.
-
-. If you are releasing a new latest version, in the `develop` branch of the `hazelcast/hazelcast-docs` repository, submit a pull request to update the `_redirects` file with your release version of IMDG.
+- In the `search-config.json` file, change the value of the `variables.version` field to your release version.
 +
-NOTE: This file is where we alias the `latest` path in URLs.
+NOTE: This file is where we specify which documentation version to index for the search.
 +
-[source,bash]
+[source,json]
 ----
-# Redirect latest imdg alias to the latest version
-/imdg/latest/*  /imdg/4.1.2/:splat 200!
+{
+  "index_name": "prod_hazelcast_docs",
+  "start_urls": [
+    {
+      "url": "https://docs.hazelcast.com/imdg/(?P<version>.*?)/",
+      "tags": [
+        "imdg"
+      ],
+      "variables": {
+        "version": ["4.2"]
+      },
 ----
 
-The documentation site will now be built, including content from your tagged version.
-
 == Major and Minor Releases
 
 This section guides you through the steps for releasing documentation for major and minor releases.
 
-. If you're writing content for a new major or minor release, make your changes in the `master` branch.
+. Create a new branch from the `master` branch. For example if you're working on content for version 5.0, create a branch named `v/5.0`.
```

Comment:
Hi @JakeSCahill , we should also document following for "Major and Minor Releases". 

Some background: In IMDG release there are two main points of the release (and there can be even couple of weeks between those points):
- create release branch - at this point the release branch is created and master is switch to the next snapshot version. Hence development can continue for master and potential issues can be fixed on release branch 
- release - release from release branch

Current docs process describes creating of `v/*` as the first step of release. However creating of `v/*` triggers build process. Hence it means one of following:
- `v/*` is created in "create release branch" stage of IMDG release. However this means that docs for `v/*` branch is live earlier than IMDG version (i.e. docs for 5.0 is live even couple of weeks before 5.0 is actually released)
- `v/*` is created in "release" stage of IMDG release. However this means that you cannot add anything what is related to IMDG `master` branch but not related to release branch until release will be done (i.e. anything can be add to `latest-dev` for even couple of weeks)

This can be fixed by adding two steps similar to IMDG release process to Docs Major and Minor Releases Release Workflow. Basically current step `Create a new branch from the master branch. For example if you’re working on content for version 5.0, create a branch named v/5.0.` could be split into two steps, something like:
1. `Create a new release branch from the master branch. For example if you’re working on content for version 5.0, create a branch named 5.0. You can add the new content (fixes) for 5.0 to this branch before 5.0 is released`.
2. `Once 5.0 should be released create a new branch from the release branch. For example if version from branch 5.0 should be released, create a branch named v/5.0.`

What do you think about it?

Reply:
I agree, thanks for the explanation. I will update the doc.